python/lsst/daf/butler/registry/_registry.py

```diff
@@ -122,19 +126,70 @@ class Registry:
     """
 
     @classmethod
-    def fromConfig(cls, config: Union[ButlerConfig, RegistryConfig, Config, str], create: bool = False,
+    def createFromConfig(cls, config: Optional[Union[RegistryConfig, str]] = None,
+                         dimensionConfig: Optional[Union[DimensionConfig, str]] = None,
+                         butlerRoot: Optional[str] = None) -> Registry:
+        """Create registry database and return `Registry` instance.
+
+        This method initializes database contents, database must be empty
+        prior to calling this method.
+
+        Parameters
+        ----------
+        config : `RegistryConfig` or `str`, optional
+            Registry configuration, if missing then default configuration will
+            be loaded from registry.yaml.
+        dimensionConfig : `DimensionConfig` or `str`, optional
+            Dimensions configuration, if missing then default configuration
+            will be loaded from dimensions.yaml.
+        butlerRoot : `str`, optional
+            Path to the repository root this `Registry` will manage.
+
+        Returns
+        -------
+        registry : `Registry`
+            A new `Registry` instance.
+        """
+        if isinstance(config, str):
```

Comment:
`RegistryConfig(None)` already acts like `RegistryConfig()` so you don't need to second guess the `RegistryConfig` constructor here.  Something like:

```python
if not isinstance(config, RegistryConfig):
    if config is None or isinstance(config, str):
        config = RegistryConfig(config)
    else:
        raise TypeError(...)
```
should work.  Makes me think that ConfigSubset could do with a classmethod for doing that check to make it consistent with what `Config` already can do.  We seem to do similar checks in fromConfig everywhere and `RegistryConfig.fromOther(config)` would be able to return either `config` or `RegistryConfig(config)` or raise.

Reply:
I'm OK with whatever you do to ConfigSubset, but in general DimensionConfig inheriting from ConfigSubset is an accident, it's only because I don't want to reimplement all good stuff in that class. Maybe we need a separate Config subclass which can be used by both ConfigSubset and DimensionConfig?

Reply:
I want to rebase our development branch ASAP before we diverge too much from master, so I'm going to merge these changes and we can work on harmonizing Config classes later.